package/lib/src/beamer_delegate.dart

```diff
@@ -869,6 +889,29 @@ class BeamerDelegate extends RouterDelegate<RouteInformation>
     return false;
   }
 
+  bool _isIntercepted(BuildContext context, BeamStack targetBeamStack) {
+    final allInterceptors = [...?parent?.interceptors, ...interceptors];
+
+    for (var i = 0; i < allInterceptors.length; i++) {
+      final interceptor = allInterceptors[i];
+      final isIntercepted = interceptor.enabled &&
+          interceptor.intercept(
+            context,
+            this,
+            _currentPages,
+            currentBeamStack,
+            targetBeamStack,
+            _deepLink,
+          );
+
+      // If any interceptor was intercepted, return true
+      if (isIntercepted) {
+        return true;
```

Comment:
As interceptor are not doing anything else except blocking the beaming, I'm thinking should we either
- rename them to `BeamBlocker` (which I don't really like as a name)
- add more functionality to them (e.g. as guards have `apply`), but then they would become almost exactly like guards :thinking: 

To discuss... I'm not sure what's the right approach here or do we even need to decide on it now.

Reply:
What if we merge guards and interceptors into one? Some thoughts on that

- I think "interceptor" is a better name so I'd like to use that. It is a breaking change, but so was `BeamStack`. It's a good time for breaking changes, now before v2
- guards can currently block, something I nearly forgot and is probably not clear, so needs improvement or documentation https://github.com/slovnicki/beamer/blob/1649a65aedf422d9d90d9692df2fa15f19db660d/package/test/beam_guard_test.dart#L438
- Runtime addition of interceptors and `BeamInterceptorScope` is an amazing addition that we definitely want to keep

What are your thoughts on merging interceptors and guards into one concept?